### ISSUE TITLE
SingletonFunction is not supported for the addLambdaFunctions method.

### ISSUE THREAD

**Issue by pradeeepst** [DataDog/datadog-cdk-constructs#194]
## Expected Behavior
I have a `singletonfunction` in my cdk code base. It would be good if  `addLambdaFunctions` could support singletonFunctions.

## Actual Behavior
Currently it supports only these - https://github.com/DataDog/datadog-cdk-constructs/blob/main/v2/src/datadog.ts#L61

Its same request as https://github.com/DataDog/datadog-cdk-constructs/issues/70

**Comment by sfirrin** [comment]
Thanks for opening @pradeeepst, if I understand that thread correctly then this should be possible for us to support now. We will look into this and keep you updated here

**Comment by joshlartz** [comment]
This is how I dealt with singletons. The trick is finding the actual function that CDK makes at the root of the stack.

```
const fn = new Function( ... );
const datadog = new Datadog(fn, 'Datadog', { ... });


// deal with singletons by finding the real function at the root of the stack
const singleton = (Stack.of(fn).node.children as Array<lambda.Function | IConstruct>).find(each => {
  const childFn = each.hasOwnProperty('functionName') ? (each as lambda.Function) : undefined;
  // the construct id param is passed down to the Function, so the physical name will be the same
  // but the final construct id will be different for the "inner" function
  return childFn && childFn.node.id !== fn.node.id && childFn.functionName === fn.functionName;
}) as lambda.Function | undefined;

datadog.addLambdaFunctions([singleton ?? (fn as lambda.Function)]);
```

**Comment by duncanpharvey** [comment]
@pradeeepst Support for Singleton functions is now available in version 1.8.1.

https://github.com/DataDog/datadog-cdk-constructs/releases/tag/v2-1.8.1
